Update advancecomp url to an actual release

kaspergrubbe opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

It does not pass `brew audit` because the SHA256 is different:

```
advancecomp:
  * stable: sha256 changed without the version also changing; please create an issue upstream to rule out malicious circumstances and to find out why the file changed.
Error: 1 problem in 1 formula detected
```

-----

# What this change does

We noticed here:  https://github.com/toy/image_optim/issues/165 that `advpng` distributed through Homebrew did not include a version, so when the documented arguments (`--version` or `-V`) were used, the program didn't write any version:

```bash
$ advpng -V
advancecomp vnone by Andrea Mazzoleni, http://www.advancemame.it
```

So we found out that Advancecomp uses a script called `autover.sh` defined here: https://github.com/amadvance/advancecomp/blob/master/autover.sh The script set a tag when building the application, it will set a tag depending on a git-tag or by reading a `.version`-file.

The source where Homebrew download the source from does not contain the `.version`-file, but the Github release page does, so when my version has been installed we get the correct output:

```bash
$ advpng -V
advancecomp v2.1 by Andrea Mazzoleni, http://www.advancemame.it
```

fxcoudert commented:
Can we use this opportunity to improve the test please?

kaspergrubbe commented:
@fxcoudert I was considering that, but was unsure of how. I would have liked to test that the version output included the version string, is that what you want? 

Do you have an example of a formula test that captures output?

fxcoudert commented:
We'd prefer a test that does something, like compress a file…

fxcoudert commented:
@BrewTestBot test this please

kaspergrubbe commented:
@fxcoudert I could make it compress a file, should I find the image and add it to the repo, or should I let the test download an image from an URL and compress that?

It seems like the tests are failing for this reason:

```
        Error: 1 problem in 1 formula detected
advancecomp:
  * stable: sha256 changed without the version also changing; please create an issue upstream to rule out malicious circumstances and to find out why the file changed.
```

Which is expected as the version hasn't changed, is there any way to get the system to disregard it?

fxcoudert commented:
Add `revision 1` line after `sha256`
For test, download a resource file and use that.

kaspergrubbe commented:
Hi @fxcoudert 

I added the revision number, and I also added a test that will compress an image, and I also added a test that tests if the output of `--version` will actually include the version number (The original issue).

Could you have a look at my changes? :)

kaspergrubbe commented:
@fxcoudert Nice fix! 

I noticed that the build still fails with the same error as before, even though I added `revision 1`, what do I need to do?

fxcoudert commented:
This is an audit warning for maintainers, we'll just go ahead and override it :)
Thanks @kaspergrubbe for your contribution to Homebrew!
